drafts/use-cases/2.api-documentation.md

```diff
@@ -0,0 +1,111 @@
+## API Documentation
+
+### Story
+As an application user
+I want my application to be able to perform some precomputations 
+So it can pre-generate some views and other UI elements to speed things up.
+
+### Usage
+```javascript
+var client = new HydraClient();
+var apiDocumentation = client.getApiDocumentation("/api");
+for (var $class of apiDocumentation.supportedClasses) {
+    // generate a partial view for the _$class_
+}
+```
+
+### Details
+The documentation obtain may be a large payload,
+thus client is expected to store it temporarely at least for the current session.
+
+```
+GET /api?documentation
+
+HTTP 200 OK
+```
+```javascript
+{
+    "@id": "/api?documentation",
```

Comment:
Well, in the light of the comment above this would be fully server driven.
First client would obtain the address of the API documentation from the base Url, and the it would use that address to obtain the API documentation.